Assets/MixedRealityToolkit/Common/Scripts/ClipPlane.cs

```diff
@@ -52,22 +103,51 @@ private void UpdatePlanePosition()
             }
         }
 
-        private void OnDrawGizmosSelected()
+        private void ToggleClippingPlane(bool clippingPlaneOn)
         {
-            if (!enabled)
+            if (Renderers == null)
             {
                 return;
             }
 
-            Gizmos.matrix = transform.localToWorldMatrix;
-            Gizmos.DrawLine(Vector3.zero, Vector3.up * 0.5f);
-            Gizmos.DrawWireCube(Vector3.zero, new Vector3(1.0f, 0.0f, 1.0f));
-        }
+            foreach (Renderer renderer in Renderers)
+            {
+                if (!renderer)
+                {
+                    continue;
+                }
 
-        private void Initialize()
-        {
-            clipPlaneID = Shader.PropertyToID("_ClipPlane");
-            materialPropertyBlock = new MaterialPropertyBlock();
+                Material material;
+
+                if(Application.isEditor && !Application.isPlaying)
+                {
+                    material = renderer.sharedMaterial;
+                }
+                else
+                {
+                    material = renderer.material;
```

Comment:
Do we dispose of this material later? 

Memory leak if not captured and disposed properly.

Reply:
Good catch! Fixed, I wish I could use material property blocks for shader keywords. :(